### PR TITLE
SM Workflow Search 6491

### PR DESCRIPTION
**Pull request by labkey-tchad**
Refactored some search and assay helpers.

### CODE REVIEW

File: src/org/labkey/test/pages/assay/ChooseAssayTypePage.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        clickAndWait(elementCache().cancelButton);
+    }
+
+    public ReactAssayDesignerPage clickNext()
+    {
+        clickAndWait(elementCache().nextButton);
+
+        return new ReactAssayDesignerPage(getDriver());
+    }
+
```
Comment (labkey-danield) [inline src/org/labkey/test/pages/assay/ChooseAssayTypePage.java]: Don't you need, or shouldn't you have, a clickNextExpectError method?

File: src/org/labkey/test/pages/assay/plate/PlateTemplateList.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.WebTestHelper;
+import org.labkey.test.pages.LabKeyPage;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class PlateTemplateList extends LabKeyPage<PlateTemplateList.ElementCache>
+{
```
Comment (labkey-danield) [inline src/org/labkey/test/pages/assay/plate/PlateTemplateList.java]: Is this class and the one above, PlateDesignerPage, used anywhere?

File: src/org/labkey/test/pages/assay/ChooseAssayTypePage.java (hunk 1 above)
Comment (labkey-tchad) [reply src/org/labkey/test/pages/assay/ChooseAssayTypePage.java]: That easy enough to add now. Didn't realize the page had any error state.

File: src/org/labkey/test/pages/assay/plate/PlateTemplateList.java (hunk 2 above)
Comment (labkey-tchad) [reply src/org/labkey/test/pages/assay/plate/PlateTemplateList.java]: Added some JavaDoc explicitly identifying these to page as incomplete.

**Pull request merged**